### ISSUE TITLE
Some refinement and reduction of Hypothesis

### ISSUE THREAD

**Issue by drvinceknight** [Axelrod-Python/Axelrod#1114]
We've been getting what feels like a few more health warning type errors with hypothesis including indications of some depreciations of features we're using (we always pip install the latest hypothesis on travis). I'll take a look at this (just opening an issue to keep track). If we get any hypothesis type failures on travis could we copy the error message in to this issue to give me some pointers :) :+1:

**Comment by drvinceknight** [comment]
At: https://travis-ci.org/Axelrod-Python/Axelrod/jobs/261514887

```
ERROR: test_property_serial_play (axelrod.tests.unit.test_tournament.TestTournament)
Test serial play using hypothesis
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Axelrod-Python/Axelrod/axelrod/tests/unit/test_tournament.py", line 221, in test_property_serial_play
    max_turns=10, min_repetitions=2,
  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/hypothesis/core.py", line 721, in wrapped_test
    random, settings, test_runner, search_strategy)
  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/hypothesis/core.py", line 398, in perform_health_checks
    HealthCheck.too_slow,
  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/hypothesis/core.py", line 272, in fail_health_check
    raise FailedHealthCheck(message)
hypothesis.errors.FailedHealthCheck: Data generation is extremely slow: Only produced 10 valid examples in 1.01 seconds (0 invalid ones and 0 exceeded maximum size). Try decreasing size of the data you're generating (with e.g.average_size or max_leaves parameters).
See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.too_slow to the suppress_health_check settings for this test.
```

**Comment by drvinceknight** [comment]
Closed by #1121
